tests/test_integration.py

```diff
@@ -0,0 +1,120 @@
+import unittest
+from collections import namedtuple
+
+import cv2
+import torch.cuda
+from kornia import image_to_tensor
+from omegaconf import OmegaConf
+from parameterized import parameterized
+from torch import Tensor
+
+from gluefactory.eval.utils import (
+    eval_homography_dlt,
+    eval_homography_robust,
+    eval_matches_homography,
+)
+from gluefactory.models.two_view_pipeline import TwoViewPipeline
+from gluefactory.settings import root
+from gluefactory.utils.image import ImagePreprocessor
+from gluefactory.utils.tensor import map_tensor
+from gluefactory.utils.tools import set_seed
+
+
+def create_input_data(cv_img0, cv_img1, device):
+    img0 = image_to_tensor(cv_img0).float() / 255
+    img1 = image_to_tensor(cv_img1).float() / 255
+    ip = ImagePreprocessor({})
+    data = {"view0": ip(img0), "view1": ip(img1)}
+    data = map_tensor(
+        data,
+        lambda t: t[None].to(device)
+        if isinstance(t, Tensor)
+        else torch.from_numpy(t)[None].to(device),
+    )
+    return data
+
+
+def squeeze_if_tensor(t):
+    return torch.squeeze(t, dim=0) if isinstance(t, Tensor) else t
+
+
+ExpectedResults = namedtuple("ExpectedResults", ("num_matches", "prec3px", "h_error"))
+
+
+class TestIntegration(unittest.TestCase):
+    methods_to_test = [
+        ("superpoint+NN.yaml", "poselib", ExpectedResults(1300, 0.8, 1.0)),
```

Comment:
Is this robust to updates in PoseLib? yes if these updates only improve the results?

Reply:
True! In general, we should resolve our dependencies in a deterministic way. And update dependencies like PoseLib consciously.

I am adding the version (hash) for PoseLib in the pyproject.toml and I am doing the same with other dependencies that can potentially suffer from the same problem (`homography_est` and `pytlsd` )